Polyfills/Canvas/Source/Colors.h

```diff
@@ -162,40 +162,52 @@ namespace Babylon::Polyfills::Internal
             {"yellow", 0xffff00},
             {"yellowgreen", 0x9acd32} };
 
-        if (str[0] == '#' && str.length() == 4)
+        if (str[0] == '#')
         {
-            unsigned int components[4];
-            int count = sscanf(str.c_str(), "#%1x%1x%1x", &components[0], &components[1], &components[2]);
-            for (int i = count; count < 4; count++)
+            unsigned int components[4] = {0xff, 0xff, 0xff, 0xff};
+            int count{};
+            int bitShift{4};
+            switch (str.length())
             {
-                components[i] += components[i] << 4;
+                case 4:
+                    count = sscanf(str.c_str(), "#%1x%1x%1x", &components[0], &components[1], &components[2]);
+                    break;
+                case 5:
+                    count = sscanf(str.c_str(), "#%1x%1x%1x%1x", &components[0], &components[1], &components[2], &components[3]);
+                    break;
+                case 7:
+                    count = sscanf(str.c_str(), "#%02x%02x%02x", &components[0], &components[1], &components[2]);
+                    bitShift = 0;
+                    break;
+                case 9:
+                    count = sscanf(str.c_str(), "#%02x%02x%02x%02x", &components[0], &components[1], &components[2], &components[3]);
+                    bitShift = 0;
+                    break;
             }
-            for (int i = count; count < 4; count++)
+            
+            for (int i = 0; i < count; i++)
             {
-                components[i] = 255;
-            }
-            return nvgRGBA(components[0], components[1], components[2], components[3]);
-        }
-        else if (str[0] == '#' && str.length() == 7)
-        {
-            unsigned int components[4];
-            int count = sscanf(str.c_str(), "#%02x%02x%02x%02x", &components[0], &components[1], &components[2], &components[3]);
-            for (int i = count; count < 4; count++)
-            {
-                components[i] = 255;
+                components[i] += components[i] << bitShift;
             }
             return nvgRGBA(components[0], components[1], components[2], components[3]);
         }
         else
         {
-            // matches strings of the form rgb(#,#,#)
-            static const std::regex rgbRegex("rgb\\(\\s*(\\d{1,3})\\s*,\\s*(\\d{1,3})\\s*,\\s*(\\d{1,3})\\s*\\)");
+            // matches strings of the form rgb(#,#,#) or rgba(#,#,#,#)
+            static const std::regex rgbRegex("rgba?\\(\\s*(\\d{1,3})\\s*,\\s*(\\d{1,3})\\s*,\\s*(\\d{1,3})\\s*\\,?\\s*(:?\\d{1,3})?\\)");
```

Comment:
I'm unclear on a few changes here:
1. Why is there an extra `\\` right before the comma? It seems like it is trying to escape a comma, which isn't needed?
2. Why is there an optional `:` before the alpha value? It almost looks like it was meant to be a non-capturing group (`?:`) except that doesn't make sense for that capture group.
3. It currently has the `,` as optional separately from the alpha value, so technically I believe the regex would match something like `rgba(1,2,3,)`, which is not idea. In this case, a non-capturing group *could* be used to group (but not capture) the `,`.
4. I think we also lost the trailing white space before the final `)`, so `rgb(1,2,3,4    )` would not match.
Given the above, I'd think it should be (untested):
```suggestion
            static const std::regex rgbRegex("rgba?\\(\\s*(\\d{1,3})\\s*,\\s*(\\d{1,3})\\s*,\\s*(\\d{1,3})\\s*(?:,\\s*(\\d{1,3}))?\\s*\\)");
```

Reply:
Fixed.